src/http/client.cr

```diff
@@ -760,7 +760,7 @@ class HTTP::Client
 
   # Closes this client. If used again, a new connection will be opened.
   def close : Nil
-    @io.try &.close
+    @io.try &.close rescue nil
     @io = nil
```

Comment:
`rescue nil` is a strong silencer. I think ensure might be more appropriate here.

```suggestion
    @io.try &.close
  ensure
    @io = nil
```

That's at least the most minimal change to fix the described behaviour.

Maybe we would want to rescue some errors, but it should probably be limited to some specific IO errors that are not worth propagating.

Reply:
Ok, then I would suggest: 

```suggestion
    @io.try &.close
  rescue IO::Error
    nil
  ensure
    @io = nil
```

When would one care of an io error when trying to close a socket?

Reply:
I actually see no strong reason to rescue any error here. If the exception bubbles up it gives the calling code an opportunity to handle it. A use case could be registering connection errors for diagnostics.

Reply:
I mean, EOFError while reading the HTTP response is silently ignored: https://github.com/crystal-lang/crystal/blob/3e6da5b8b8a7599ae745b9f1747d69c8fc10c7f1/src/http/client/response.cr#L127 
Inconsistent and not very user friendly. I argue that requests should be retried silently once for both read (already done) and write errors.

Reply:
When the user intends to close the connection, why would it care that the connection might already be broken?

Reply:
No exception is raised if @io is nil, why wouldn't one then want to "register" that for diagnostics? What's the difference for that against a broken or already closed socket?